perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -2054,6 +2056,54 @@ public Application getApplicationById(PerunSession sess, int appId) throws Regis
 
 	}
 
+	@Override
+	public Paginated<Application> getApplicationsPage(PerunSession userSession, Vo vo, ApplicationsPageQuery query) throws PerunException {
+		vosManager.checkVoExists(userSession, vo);
+		MapSqlParameterSource namedParams = new MapSqlParameterSource();
+
+		//Authorization
+		if (query.getGroupId() != null && !AuthzResolver.authorizedInternal(userSession, "getApplicationsForGroup_Group_List<String>_policy", Collections.singletonList(groupsManager.getGroupById(userSession, query.getGroupId())))) {
+			throw new PrivilegeException(userSession, "getApplicationsForGroup");
+		} else if (!AuthzResolver.authorizedInternal(userSession, "getApplicationsForVo_Vo_List<String>_Boolean_policy", Collections.singletonList(vo))) {
+			throw new PrivilegeException(userSession, "getApplicationsForVo");
+		}
+
+
+		namedParams.addValue("voId", vo.getId());
+		namedParams.addValue("states", query.getStates());
+		namedParams.addValue("dateFrom", query.getDateFrom());
+		namedParams.addValue("dateTo", query.getDateTo());
+		namedParams.addValue("offset", query.getOffset());
+		namedParams.addValue("limit", query.getPageSize());
+		namedParams.addValue("memberId", query.getMemberId());
+		namedParams.addValue("groupId", query.getGroupId());
+		String searchQuery = getSQLWhereForApplicationsPage(query, namedParams);
+
+		Paginated<Application> applications = namedJdbc.query(
+			APP_SELECT_PAGE +
+				" WHERE a.vo_id=(:voId)" +
+				(query.getStates() == null || query.getStates().isEmpty() ? "" : " AND a.state IN (:states) ") +
+				(query.getIncludeGroupApplications() ? "" : " AND a.group_id is null") +
+				(query.getMemberId() == null ? "" : "  AND a.user_id=(:memberId)") +
```

Comment:
MemberId shouldn't be matched to userId, userId should be obtained by calling `getMemberById`and retrieving its userId, which can then be used in the WHERE clause. It will need fixing in the tests then too.

Reply:
Changed the name of the attribute in ApplicationPageQuery to userId, as this was what I wanted it to represent in the first place. When searching for applications a member of a group/vo has in that group/vo, it doesn't matter whether we use his memberId or userId, does it? And as his userId is stored in the database with the applications, I believe this to be the simpler solution.